app/javascript/entrypoints/pdc/work_readme_file_upload.es6

```diff
@@ -0,0 +1,73 @@
+/* global Uppy */
+export default class WorkReadmeFileUpload {
+  constructor(triggerButtonId, uppyAreaId) {
+    this.triggerButtonId = triggerButtonId;
+    this.uppyAreaId = uppyAreaId;
+  }
+
+  attach_validation() {
+    const uploadUrl = $('#uppy_upload_url').text();
+    if (uploadUrl !== '') {
+      WorkReadmeFileUpload.setupUppy(this.triggerButtonId, this.uppyAreaId, uploadUrl);
+    }
+  }
+
+  // Setup Uppy to handle file uploads
+  // References:
+  //    https://uppy.io/docs/quick-start/
+  //    https://davidwalsh.name/uppy-file-uploading
+  static setupUppy(triggerButtonId, uppyAreaId, uploadUrl) {
+    // https://uppy.io/blog/2018/08/0.27/#autoproceed-false-by-default
+    // https://uppy.io/docs/uppy/#restrictions
+    const uppy = Uppy.Core({
+      autoProceed: true,
+      restrictions: {
+        maxNumberOfFiles: 1,
+        allowedFileTypes: ['.txt', '.md'],
+      },
+      onBeforeUpload(files) {
+        // source: https://github.com/transloadit/uppy/issues/1703#issuecomment-507202561
+        if (Object.entries(files).length === 1) {
+          const file = Object.entries(files)[0][1];
+          if (file.meta.name.toLowerCase().includes('readme') === true) {
+            // we are good
+            return true;
+          }
+        }
+        uppy.info('You must select a file that includes the word README in the name', 'error');
+        return false;
+      },
+    });
+
+    // Configure the initial display (https://uppy.io/docs/dashboard)
+    uppy.use(Uppy.Dashboard, {
+      target: `#${uppyAreaId}`,
+      inline: false, // display of dashboard only when trigger is clicked
+      trigger: `#${triggerButtonId}`,
+    });
+
+    // We use the XHRUploader, this is the most basic uploader (https://uppy.io/docs/xhr-upload/)
+    // X-CSRF-Token: https://stackoverflow.com/a/75050497/446681
+    const token = document.querySelector("meta[name='csrf-token']");
+    let tokenContent;
+    if (token) {
+      tokenContent = token.content;
+    } else {
+      tokenContent = '';
+    }
+    uppy.use(Uppy.XHRUpload, {
+      endpoint: uploadUrl,
+      headers: { 'X-CSRF-Token': tokenContent },
+      bundle: true, // upload all selected files at once
```

Comment:
It probably does not matter. Setting it to true made the request submit the file identically to the way our previous uploader did: always as an array. In this case it will always be an array of 1 file so I guess we could go either way but I tried to keep the code as compatible with the previous one.